polaris.shopify.com/src/scripts/og-images/index.js

```diff
@@ -0,0 +1,165 @@
+  family: "Inter",
+});
+
+const CANVAS_WIDTH = 1200;
+const CANVAS_HEIGHT = 630;
+const CANVAS_MARGIN = 60;
+
+const canvas = createCanvas(CANVAS_WIDTH, CANVAS_HEIGHT);
+const ctx = canvas.getContext("2d");
+
+async function createImage(url) {
+  let charsOnLine = 0;
+  const title = getTitleForUrl(url)
+    .split(" ")
+    .map((word) => {
+      if (charsOnLine + word.length >= 12) {
+        return `\n${word}`;
+        charsOnLine = 0;
+      }
+      charsOnLine += word.length;
+      return word;
+    })
+    .join(" ");
+
+  // Draw the background
+  ctx.fillStyle = "rgb(50,50,50)";
+  ctx.fillRect(0, 0, CANVAS_WIDTH, CANVAS_HEIGHT);
+
+  // Add component images
+  if (url.startsWith("/components")) {
+    const urlSegments = url.split("/");
+    const slug = urlSegments[urlSegments.length - 1];
+    const previewImagePath = `./public/component-previews/${slug}.png`;
+    const previewExists = fs.existsSync(previewImagePath);
+
+    if (previewExists) {
+      const imageRect = {
+        x: 600,
+        y: 0,
+        width: 1050,
+        height: 600,
+      };
+
+      await loadImage(previewImagePath).then((image) => {
+        ctx.drawImage(
+          image,
+          imageRect.x,
+          imageRect.y,
+          imageRect.width,
+          imageRect.height
+        );
+      });
+
+      // Draw black over the image to darken it
+      ctx.fillStyle = "rgba(0,0,0,.8)";
+      ctx.fillRect(imageRect.x, imageRect.y, imageRect.width, imageRect.height);
+    }
+  }
+
+  // Draw the title
+  ctx.fillStyle = "#fff";
+  ctx.font = "80px Inter";
+  ctx.fillText(title, CANVAS_MARGIN, CANVAS_MARGIN + 80);
+
+  // Draw Shopify logo
+  await loadImage("./src/scripts/og-images/shopify_glyph_white.png").then(
+    (image) => {
+      const logoDimensions = {
+        width: 913,
+        height: 1035,
+      };
+
+      const logoScale = 0.04;
+
+      ctx.drawImage(
+        image,
+        CANVAS_MARGIN,
+        CANVAS_HEIGHT - CANVAS_MARGIN - logoDimensions.height * logoScale,
+        logoDimensions.width * logoScale,
+        logoDimensions.height * logoScale
+      );
+
+      // Write "Polaris"
+      ctx.fillStyle = "#fff";
+      ctx.font = "26px Inter";
+      ctx.fillText(
+        "Polaris",
+        CANVAS_MARGIN + logoDimensions.width * logoScale + 20,
+        CANVAS_HEIGHT - CANVAS_MARGIN - 10
+      );
+    }
+  );
+
+  ctx.fillStyle = writeImage(canvas, url);
+}
+
+function writeImage(canvas, url) {
+  fs.writeFileSync(
+    `./public/open-graph/${
+      slugify(url.replace("/", "").replace(/\//g, "--")) || "home"
+    }.jpg`,
+    canvas.toBuffer()
+  );
+}
+
+function getTitleForUrl(url) {
+  let title = "Polaris";
+  const urlSegments = url.split("/");
+  const slug = capitalizeFirstLetter(urlSegments[urlSegments.length - 1]);
+
+  title = slug.replace(/-/g, " ");
+
+  if (url.startsWith("/tokens/")) {
+    title = `${title} tokens`;
+  }
+
+  if (title === "") {
+    title = "Polaris";
+  }
+
+  return title;
```

Comment:
Uhmmm in line 138 here we are reassigned the value ("Polaris") that we have already assigned in line 127

I think we can use the `slug` instead of `title` in this "if" statement and then we do not need to do line 131 or reassign "Polaris" to title, something like this:

```suggestion
  let title = "Polaris";
  const urlSegments = url.split("/");
  const slug = capitalizeFirstLetter(urlSegments[urlSegments.length - 1]).replace(/-/g, " ");

  if (slug) {
    const isToken = url.startsWith("/tokens/")
    title = `${slug}${isToken && ' tokens'}`;
  }

  return title;
```

What do you think?

Reply:
Thanks! I took some hints from what you suggested and rewrote it a bit. 

The only reason I want to keep the...

```
 if (url.startsWith("/tokens/")) {
    title = `${title} tokens`;
  }
```

...around is that it's easy to just add more `if`s down the road if we want more title hacks. 🤷‍♀️

Let's clean this up more down the road! 👀